zerver/lib/zcommand.py

```diff
@@ -73,4 +77,52 @@ def change_mode_setting(
                 setting_value=False,
             )
         )
+    elif command == "digress":
+        data_keys = [
+            "old_stream",
+            "old_topic",
+            "new_stream",
+            "new_topic",
+        ]
+        if not data or not all(key in data for key in data_keys):
+            raise JsonableError(_("Invalid data."))
+
+        if data["old_stream"] == data["new_stream"] and data["old_topic"] == data["new_topic"]:
+            return dict(msg=_("Cannot digress to the same thread."))
+
+        (old_stream, ignored_old_stream_sub) = access_stream_by_name(
+            user_profile, data["old_stream"]
+        )
+
+        (new_stream, ignored_new_stream_sub) = access_stream_by_name(
+            user_profile, data["new_stream"]
+        )
+
+        old_topic_exists = user_message_exists_for_topic(
+            user_profile=user_profile,
+            recipient_id=old_stream.recipient_id,
+            topic_name=data["old_topic"],
+        )
```

Comment:
This check deserves at least a comment, because it's not strictly correct -- a user could in theory be trying to digress in a public stream they're not subscribed to and thus don't have any UserMessage objects for.  

I think the fix would be to check if the user has access to full history for the stream via the standard helper, and in that case, do effectively `message_exists_for_topic` instead.

If it's easy which I think it is, best to just fix, but if it's hard, we can open a follow-up issue and leave a comment here.